src/confd/src/infix-dhcp-client.c

```diff
@@ -214,9 +232,24 @@ static void add(const char *ifname, struct lyd_node *cfg)
 		snprintf(do_arp, sizeof(do_arp), "-a%d", ARPING_MSEC);
 
 	if (client_id && client_id[0]) {
-		args = alloca(strlen(client_id) + 12);
-		if (args)
-			sprintf(args, "-C -x 61:'\"%s\"'", client_id);
+		size_t len = 3 * strlen(client_id) + 16;
+
+		cid = malloc(len);
+		if (!cid)
+			goto generr;
+
+		strlcpy(cid, "-C -x 61:00", len);
+		if (is_hex(client_id)) {
```

Comment:
Are we not missing something in the model that tells us whether the user wants to use a string or a binary client-id?

Example: I have 10 devices. I configure them to send `D${N}` as their client ID. D1 through D9 will then, IIUC, send `61:00:d<1-9>`, whereas D10 will send `61:00:44:31:30`.

If the `client-id` leaf was turned into a container that contained a choice between `leaf string { type string }` and a `leaf octets { type string; pattern '...'; }`, we could parse the example above unambiguously.

Reply:
Just to be clear, a string "d10" would be translated properly in the current regime, the `is_hex()` function only classifies proper pairs of hexadecimal octets separated with ':' as hexadecimal.   But yeah, as you mention in your next comment, we could create a shared grouping of this for both server and client.

Reply:
Yes, "d10" would work, but "d9" would be interpreted as a single octet. ...but as you say, all of that is mute if we go for the shared datatype from the server.

Reply:
Please see the updated models in this latest (final) update.